src/analysis/syntax_doc.ml

```diff
@@ -0,0 +1,16 @@
+type info = {
+  name: string;
+  description: string;
+  example: string;
+  documentation: string;
+}
+
+let node_infos : info list = [
```

Comment:
The more I look at it, the more I am convinced that we don't need this intermediate database since we have no other usage for it than the syntax-doc command.

Let's simply do the matching here, and return the records directly:

```
match node with
| a_type_extension -> { name="Type extension"; description...}
l ... -> ...
```